### PR TITLE
Copy quantization table in zig zag order

### PR DESCRIPTION
**Pull request by toffe**
The JPEG specification requires the DQT to be in zig zag order and I have
verified with Freescale (NXP) that the VPU also expects this order.

### CODE REVIEW

**Comment by toffe** [comment]
This commit solves issue #14.

File: imxvpuapi/imxvpuapi_vpulib.c [hunk 1]
```diff
@@ -2629,10 +2644,10 @@ static void imx_vpu_enc_set_mjpeg_tables(unsigned int quality_factor, EncMjpgPar
 	else
 		scale_factor = 200 - quality_factor * 2;
 
-	imx_vpu_enc_copy_quantization_table(mjpeg_params->qMatTab[DC_TABLE_INDEX0], mjpeg_enc_quantization_luma,   sizeof(mjpeg_enc_quantization_luma),   scale_factor);
```
Comment (dv1) [inline imxvpuapi/imxvpuapi_vpulib.c]: Ah, this is an interesting. It does contradict the indices from the imx-vpu vpu_lib.h header:

    #define DC_TABLE_INDEX0		    0
    #define AC_TABLE_INDEX0		    1
    #define DC_TABLE_INDEX1		    2
    #define AC_TABLE_INDEX1		    3

but these indices themselves do not seem to apply for MJPEG. From looking at the JpgDecQMatTabSetUp() function inside imx-vpu's vpu_util.c function, your change seems to be correct. Is this how you've noticed that the order was wrong?

**Comment by dv1** [comment]
Anyway, your change works for me. Thanks for that, finally fixing this was on my list :)

File: imxvpuapi/imxvpuapi_vpulib.c (hunk 1 above)
Comment (toffe) [inline imxvpuapi/imxvpuapi_vpulib.c]: I studied the definition of qMatTab in the VPU API reference manual. It specifies the offsets for the Y, Cb and Cr quantization tables. I also looked at the component info tables, and as stated in the comment "The fourth byte is the index of the quantization table to use" there it is specified which qMatTab indices we are actually using. Index 1 is always specified for both Cb and Cr so we really only need to specify 0 and 1, but I kept index 2 for future reference.

**Pull request merged**